intents/show.py

```diff
@@ -71,34 +71,85 @@ def show(table,**kwargs):
         given `table(a pandas dataframe object)``
 
     """
-    
-	date_column_name = kwargs.get('date_column_name', 'date')
-	date_range = kwargs.get('date_range', None)
-	date_format = kwargs.get('date_format', 'yyyy-mm-dd')
 
-	slices = kwargs.get('slices', None)
+    date_column_name = kwargs.get('date_column_name', 'date')
+    date_range = kwargs.get('date_range', None)
+    date_format = kwargs.get('date_format', 'yyyy-mm-dd')
 
-	summary_operator = kwargs.get('summary_operator', None)
+    slices = kwargs.get('slices', None)
 
-	metric = kwargs.get('metric',None)
+    summary_operator = kwargs.get('summary_operator', None)
 
-	dimensions = kwargs.get('dimensions',None)
+    metric = kwargs.get('metric',None)
 
-	table = aspects.apply_date_range(table, date_range,date_column_name, date_format)
+    dimensions = kwargs.get('dimensions',None)
 
-	table = aspects.slice_table(table, slices)
+    table = aspects.apply_date_range(table, date_range,date_column_name, date_format)
 
-	# collecting the colums not to be removed
-	required_columns = []
-	if dimensions is not None:
-		required_columns = dimensions.copy()
-	# metric is optional in show	
-	if metric is not None:
-		required_columns.append(metric)
+    table = aspects.slice_table(table, slices)
 
-	table = aspects.crop_other_columns(table, required_columns)
+    # collecting the colums not to be removed
+    required_columns = []
+    if dimensions is not None:
+    	required_columns = dimensions.copy()
+    # metric is optional in show	
+    if metric is not None:
+        required_columns.append(metric)
 
-	table = aspects.group_by(table, dimensions, summary_operator)
+    table = aspects.crop_other_columns(table, required_columns)
 
-	return table
-    
+    # When there is no dimension to group by but a summary_operator exists ,
+    # we assume that user wants to apply the summary operator on whole data
+    if( (dimensions is None) and (summary_operator is not None) ):
+        # We add temporary column of 'Summary Operator' by which we can groupby
+        # to obtain the final result
+        add_temporary_column_of_summary_operator(table,summary_operator)
+
+        dimensions = []
+
+        # To groupby 'Summary Operator' column inserted
+        dimensions.append('Summary Operator')
+
+    table = aspects.group_by(table, dimensions, summary_operator)
+
+    return table
+
+
+
+
+def add_temporary_column_of_summary_operator(table,summary_operator):
```

Comment:
Since this is an internal function, prefix name with _

Reply:
Updated